### PR TITLE
feat(mesh): implement pick button with pub/sub api

### PR DESCRIPTION
**Pull request by tyc67**
## Notable Changes

- add mutation method into `StoryPickButton` for real-time updates, following pub/sub [ documentation ](https://paper.dropbox.com/doc/Mesh-PubSub-requirement--CRYm15Qcszsoi36jiQTOlp0_Ag-n67xllrJ6U60gBfczXpQN)
- revise graphql query documents to include `pick.kind` and `pick.is_active` as the [ spec ](https://paper.dropbox.com/doc/Mesh-PubSub-requirement--CRYm15Qcszsoi36jiQTOlp0_Ag-n67xllrJ6U60gBfczXpQN)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207450819809545

### CODE REVIEW

File: packages/mesh/components/story-card/story-pick-button.tsx [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
 export default function StoryPickButton({
   isStoryPicked,
+  storyId,
 }: {
   isStoryPicked: boolean
+  storyId: string
 }) {
   const [isPicked, setIsPicked] = useState(isStoryPicked)
+  const params = useParams()
+  const { id: memberId } = params
```
Comment (erase2004) [inline packages/mesh/components/story-card/story-pick-button.tsx]: 由於該元件放在 `components` 底下，在其他頁面被使用到時，id 資訊未必就是會員 id，因此需要 `TODO: ` 來標註，待會員登入實作後，改從會員資訊的 store 獲取對應資訊

File: packages/mesh/components/story-card/story-pick-button.tsx [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
 
-  const handleClickPick = () => {
+  const handleClickPick = async () => {
```
Comment (erase2004) [inline packages/mesh/components/story-card/story-pick-button.tsx]: 建議加上 debounce

File: packages/mesh/components/story-card/story-pick-button.tsx [hunk 3]
```diff
@@ -1,23 +1,40 @@
...
-      //TODO: handleDeletePick()
+
+    const payload = {
+      action: isPicked ? 'remove_pick' : 'add_pick',
+      memberId,
+      objective: 'story',
+      targetId: storyId,
+      ...(!isPicked && { state: 'public' }),
     }
+    const response = await fetchData(RESTFUL_ENDPOINTS.pubsub, {
```
Comment (erase2004) [inline packages/mesh/components/story-card/story-pick-button.tsx]: 這其實應該是 sendData 的行為，但目前沒有相關名稱的方法，如果要增加的話，似乎又會與 https://github.com/readr-media/Sachiel/pull/780 的修改內容產生衝突

Comment (tyc67) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 也許要有 queryFn 和 mutationFn ?

File: packages/mesh/components/story-card/story-pick-button.tsx (hunk 1 above)
Comment (tyc67) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 修改於 [b9b8caf](https://github.com/readr-media/Sachiel/pull/781/commits/b9b8caf7cf450f007341e532c8e1cc2e4a6c16c8)

File: packages/mesh/components/story-card/story-pick-button.tsx (hunk 2 above)
Comment (tyc67) [reply packages/mesh/components/story-card/story-pick-button.tsx]: - 修改於 [b9b8caf](https://github.com/readr-media/Sachiel/pull/781/commits/b9b8caf7cf450f007341e532c8e1cc2e4a6c16c8), 
- 改用server actions [6be4746](https://github.com/readr-media/Sachiel/pull/781/commits/6be4746fb4b633bd5491801a732ebeda8ac920b2)
- `action.ts` 改至 `/components` [1dcc4f5](https://github.com/readr-media/Sachiel/pull/781/commits/1dcc4f562a8855cdfb790797cbb62dad120e7060)
- TODO: toast message [0ee5502](https://github.com/readr-media/Sachiel/pull/781/commits/0ee55021f0375a63008fd7cd9fe61c77fcdb5fe2)

File: packages/mesh/components/story-card/story-pick-button.tsx (hunk 3 above)
Comment (caesarWHLee) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 如果 #780 的做法有取得共識我再來更新這邊的 conflict

Comment (caesarWHLee) [reply packages/mesh/components/story-card/story-pick-button.tsx]: `setIsPicked(!isPicked)`  應該是放到 await fetchData 後才修改，避免網路問題連 endpoint 都沒摸到就顯示精選成功了

File: packages/mesh/app/social/[id]/page.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ import Feed from '../_components/feed'
 import FollowSuggestionFeed from '../_components/follow-suggestion-feed'
 import FollowSuggestionWidget from '../_components/follow-suggestion-widget'
 
-export const revalidate = 60
+export const revalidate = 0
```
Comment (caesarWHLee) [inline packages/mesh/app/social/[id]/page.tsx:14]: (optional) 看要不要加個 TODO comment 來提醒我們之後要回來處理 cache 的問題

File: packages/mesh/components/story-card/story-pick-button.tsx (hunk 3 above)
Comment (tyc67) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 目前 api 沒有 loading state, 放在前後會有差異嗎?

Comment (caesarWHLee) [reply packages/mesh/components/story-card/story-pick-button.tsx]: 放在後面的差別主要是 fetch failed 的狀況就不應該把 isPicked 設定為 !isPicked

Comment (tyc67) [reply packages/mesh/components/story-card/story-pick-button.tsx]: - 修改於 [dbd63e2](https://github.com/readr-media/Sachiel/pull/781/commits/dbd63e29ef07615717ddfeb50e63bf66850550b6), 
- 改用server actions [6be4746](https://github.com/readr-media/Sachiel/pull/781/commits/6be4746fb4b633bd5491801a732ebeda8ac920b2), 
- `action.ts` 改至 `/components` [1dcc4f5](https://github.com/readr-media/Sachiel/pull/781/commits/1dcc4f562a8855cdfb790797cbb62dad120e7060)
- TODO: toast message [0ee5502](https://github.com/readr-media/Sachiel/pull/781/commits/0ee55021f0375a63008fd7cd9fe61c77fcdb5fe2)

File: packages/mesh/app/social/[id]/page.tsx (hunk 4 above)
Comment (tyc67) [reply packages/mesh/app/social/[id]/page.tsx:14]: 修改於 [a8acefe](https://github.com/readr-media/Sachiel/pull/781/commits/a8acefe9f7b5cd4da91ca35b40274451e883458a)

**Comment by caesarWHLee** [comment]
LGTM 👍

File: packages/mesh/components/action.ts
Comment (erase2004) [inline packages/mesh/components/action.ts:1]: 應該用 `fetchRestful` 而不是 `fetchStatic` 的方法，畢竟行為上是 call RESTful API，而不是抓取 json 之類的靜態檔案?

Comment (caesarWHLee) [reply packages/mesh/components/action.ts:1]: 建議在 pacakge root 新增 actions 資料夾，這個檔名可以改成 pick.ts 專門處理 pick 相關的 action

Comment (erase2004) [reply packages/mesh/components/action.ts:1]: convention 目前沒有，你們可以討論後再更新上去

Comment (tyc67) [reply packages/mesh/components/action.ts:1]: 修改於 [0c4cfa9](https://github.com/readr-media/Sachiel/pull/781/commits/0c4cfa95c323e71ca21067a2387345426cb7181d)

**Review by caesarWHLee — APPROVED** [review]
LGTM 🚀

**Review by erase2004 — APPROVED** [review]
LGTM 👏

**Pull request merged**